plugins/woocommerce/client/admin/client/marketplace/components/product-preview-modal/product-preview-modal.tsx

```diff
@@ -0,0 +1,191 @@
+		html: string;
+		css: string;
+	} | null >( null );
+	const [ error, setError ] = useState< string | null >( null );
+
+	const closeModal = useCallback(
+		( closeType = '' ) => {
+			if ( onClose ) {
+				onClose( closeType );
+			}
+			// Return focus to the triggering element after a small delay
+			// to ensure layout shifts have completed
+			setTimeout( () => {
+				if ( triggerRef.current ) {
+					triggerRef.current.focus();
+				}
+			}, 100 );
+		},
+		[ onClose, triggerRef ]
+	);
+
+	// Add event listener for content interactions
+	useEffect( () => {
+		const handleContentInteraction = ( event: Event ) => {
+			const target = event.target as HTMLElement;
+			const link = target.closest( 'a' );
+			if ( link ) {
+				const trackType = link.getAttribute( 'data-iam-tracks' );
+				if ( trackType === 'buy_now' ) {
+					closeModal(
+						'marketplace_product_preview_modal_buy_now_clicked'
+					);
+				}
+
+				if ( trackType === 'see_more' ) {
+					closeModal(
+						'marketplace_product_preview_modal_see_more_clicked'
+					);
+				}
+			}
+		};
+
+		const contentElement = document.querySelector(
+			'.woocommerce-marketplace__product-preview-modal__content'
+		);
+		if ( contentElement ) {
+			contentElement.addEventListener(
+				'click',
+				handleContentInteraction
+			);
+			return () => {
+				contentElement.removeEventListener(
+					'click',
+					handleContentInteraction
+				);
+			};
+		}
+	}, [ closeModal ] );
+
+	// Fetch preview content and record event when the modal mounts
+	useEffect( () => {
+		const loadPreview = async () => {
+			try {
+				const response = await fetchProductPreview( productId );
+				const previewData = response?.data || response;
+
+				if ( ! previewData?.html || ! previewData?.css ) {
+					throw new Error( 'Invalid preview data structure' );
+				}
+
+				const sanitizedHtmlObj = sanitizeHtmlExtended(
+					previewData.html,
+					sanitizeHtmlConfig
+				) as { __html?: string };
+				const sanitizedHtml = sanitizedHtmlObj?.__html ?? '';
+
+				setPreviewContent( {
+					html: sanitizedHtml,
+					css: previewData.css,
+				} );
+				setError( null );
+			} catch ( err ) {
+				setError(
+					__( 'Failed to load product preview.', 'woocommerce' )
+				);
+			} finally {
+				setIsLoading( false );
+			}
+		};
+
+		loadPreview();
+		if ( onOpen ) {
+			onOpen();
+		}
+	}, [ onOpen, productId ] );
+
+	const productHeader = (
+		<div className="woocommerce-marketplace__product-preview-modal__header">
+			{ productIcon && (
+				<img
+					className="woocommerce-marketplace__product-preview-modal__icon"
+					src={ productIcon }
+					alt={ productTitle }
+				/>
+			) }
+			<div className="woocommerce-marketplace__product-preview-modal__header-content">
+				<h2>{ productTitle }</h2>
+				{ productVendor && (
+					<div className="woocommerce-marketplace__product-preview-modal__vendor">
+						<span>{ __( 'By ', 'woocommerce' ) }</span>
+						{ productVendor }
```

Comment:
Do we need to keep space in the translated string or should we rather use something like this?
```suggestion
						<span>{ __( 'By', 'woocommerce' ) }</span> { productVendor }
```

Reply:
The linter doesn't like your implementation, but nothing wrong with the idea. Added a linter-approved alternative in bf66c9e3ff